R/dindep.R

```diff
@@ -0,0 +1,30 @@
+#' @title density of the independence copula
+#' @description identifies the density of the independence copula,
+#' 1 everywhere on the unit square, and 0 elsewhere
+#'
+#' @param vector_1 A numeric vector
+#'
+#' @param vector_2 A numeric vector
+#' @return A numeric vector
+#' @examples
+#' dindep(c(1, 0.3, 5, 0.8), c(0.5, 0.7, 3, 8))
+#'
+#' @export
+
+dindep <- function(vector_1, vector_2) {
+
+  if ((length(vector_1) == 1) | (length(vector_2) == 1)) {
+    vector_1 <- vctrs::vec_recycle_common(vector_1, vector_2)[[1]]
+    vector_2 <- vctrs::vec_recycle_common(vector_1, vector_2)[[2]]
+  }
+  if ((length(vector_1) == 0) | (length(vector_2) == 0)) {
+    return(0)
+  }
+
+  if (length(vector_1) != length(vector_2)) {
+    stop("Unable to operate on the vectors of a different length if none of them is of length 1 ")
+  }
+  output_vector <- between(vector_1, 0, 1) & between(vector_2, 0, 1)
```

Comment:
Interesting! I never knew about this `between()` function. From dplyr? One important thing about making an R package is to depend on as few other R packages as possible, so that there are less moving parts that can break the package. We should avoid using packages from the tidyverse, unless it's a "bonus" (such as outputting a tibble instead of a data frame if a user has the tibble package installed).